### PR TITLE
[SPARK-32890][SQL] Pass all `sql/hive` module UTs in Scala 2.13

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr fix failed cases in sql hive module in Scala 2.13 as follow:

- HiveSchemaInferenceSuite (1 FAILED -> PASS)
- HiveSparkSubmitSuite (1 FAILED-> PASS)
- StatisticsSuite (1 FAILED-> PASS)
- HiveDDLSuite (1 FAILED-> PASS)

After this patch all test passed in sql hive module in Scala 2.13.

### Why are the changes needed?
We need to support a Scala 2.13 build.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
- Scala 2.12: Pass the Jenkins or GitHub Action

- Scala 2.13: All tests passed.

Do the following:

```
dev/change-scala-version.sh 2.13
mvn clean install -DskipTests -pl sql/hive -am -Pscala-2.13 -Phive
mvn clean test -pl sql/hive -Pscala-2.13 -Phive
```

**Before**

```
Tests: succeeded 3662, failed 4, canceled 0, ignored 601, pending 0
*** 4 TESTS FAILED ***
```

**After**

```
Tests: succeeded 3666, failed 0, canceled 0, ignored 601, pending 0
All tests passed.
```

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @srowen , this pr fix all failed test cases in `sql/hive` module. In this patch add `test-2.13.jar` to `/Users/yangjie/SourceCode/git/spark-mine/sql/hive/src/test/resources/regression-test-SPARK-8489` path, the `test-2.13.jar` not a copy of `test-2.12.jar`, I try to generate `test-2.13` as https://github.com/apache/spark/pull/22308 but failed because there is no spark 3.1.0-SNAPSHOT in maven repository, and I generate it as follow:

1. create a `test` module in spark source
2. copy `Main.scala` and `MyCoolClass.scala` to `src/main/scala` dir of `test` module
3. create a `pom.xml` use spark base `pom.xml`  as parent pom and add dependency 
```
    <dependency>
      <groupId>org.apache.spark</groupId>
      <artifactId>spark-sql_${scala.binary.version}</artifactId>
      <version>${project.version}</version>
    </dependency>
```
4. mvn build `test` module with -am and rename `original-test-2.13-3.1.0-SNAPSHOT.jar` to `test-2.13.jar`

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSchemaInferenceSuite.scala [hunk 1]
```diff
@@ -129,7 +129,8 @@ class HiveSchemaInferenceSuite
     // properties out).
     assert(!externalCatalog.getTable(DATABASE, TEST_TABLE_NAME).schemaPreservesCase)
     val rawTable = client.getTable(DATABASE, TEST_TABLE_NAME)
-    assert(rawTable.properties.filterKeys(_.startsWith(DATASOURCE_SCHEMA_PREFIX)) == Map.empty)
+    assert(
```
Comment (srowen) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSchemaInferenceSuite.scala:132]: Can we just check isEmpty?

**Review by srowen — COMMENTED** [review]
Seems OK to me

**Comment by srowen** [comment]
Jenkins test this please

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSchemaInferenceSuite.scala (hunk 1 above)
Comment (LuciferYang) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSchemaInferenceSuite.scala]: Address ce5c81b fix this

Comment (LuciferYang) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveSchemaInferenceSuite.scala]: check isEmpty looks simpler

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**